app/models/compute_resource.rb

```diff
@@ -123,6 +124,10 @@ def to_label
     "#{name} (#{provider_friendly_name})"
   end
 
+  def connection_options
+    http_proxy ? {:proxy => http_proxy.full_url} : {}
```

Comment:
By doing this, we would no longer be supporting the environment variables, correct?   I would be supportive of that!